### ISSUE TITLE
[BUG] Unlike its pandas counterpart, `cudf.date_range` doesn't include the end if it's specified

### ISSUE THREAD

**Issue by amanlai** [rapidsai/cudf#15116]
If I create a DatetimeIndex object using `cudf.date_range()` by passing `start`, `end` and `freq` parameters:
```python
cudf.date_range('2020-01-01', '2020-01-05', freq='D')

DatetimeIndex(['2020-01-01', '2020-01-02', '2020-01-03', '2020-01-04'], 
              dtype='datetime64[ns]', freq='D')
```
As you can see, the result is an object with length 4. The same code in pandas results in an object with length 5:
```python
pd.date_range('2020-01-01', '2020-01-05', freq='D')

DatetimeIndex(['2020-01-01', '2020-01-02', '2020-01-03', '2020-01-04', '2020-01-05'],
              dtype='datetime64[ns]', freq='D')
```

Is it possible to make it consistent with pandas' `date_range()` in that the result includes both ends? Because, `closed` parameter is nonfunctional at the moment, to make `cudf.date_range()` produce the same output as its pandas counterpart, we need to add 1 `freq` to `end`, which creates a whole lot of work: convert the string to datetime, convert freq to timedelta and add them to define a new end.

Anyway, is it possible to make it clear in the documentation that this is different from its pandas counterpart?

<sup>I didn't know how to tag this since it's not really a bug. It's just different from pandas API that results in a subtle bug _in my code_.</sup>

**Comment by shwina** [comment]
I think this is a legitimate bug. Thanks for reporting!

**Comment by Matt711** [comment]
Was this closed by #16277?

cc. @mroeschke

**Comment by mroeschke** [comment]
> Was this closed by [#16277](https://github.com/rapidsai/cudf/pull/16277)?
> 
> cc. [@mroeschke](https://github.com/mroeschke)

Appears so! Closing
